### PR TITLE
Stats: Traffic page header refresh - sticky header

### PR DESCRIPTION
**Pull request by annacmc**
Related to https://github.com/Automattic/red-team/issues/217

## Proposed Changes

* Makes the header on the traffic page sticky

## Why are these changes being made?

* Part of the Date filtering for the traffic page project (header refresh section)

## Important Notes

* Initially explored by @a8ck3n in https://github.com/Automattic/wp-calypso/pull/95573 - it's important to note that this requires a change to the `layout__content` DIV which effects the entire viewport. However I have added in `&:has(.stats__period-header.is-sticky)` to the selector, which hopefully will prevent unintended issues when the sticky header is not being used. But we do want to be really thorough in testing this. 

## Testing Instructions

* Open the Calypso Live link.
* Visit the Stats → Traffic page and confirm it loads normally.
* Manually enable the feature flag: ?flags=stats/new-date-filtering
* Scroll the page and confirm the header sticks to the top of the page.
* Keep an eye out for unintended layout changes.

| Design Image | Second Header |
| ------------- | ------------- |
| image  | Content Cell  |
| Content Cell  | Content Cell  |

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-period-header/index.jsx [hunk 1]
```diff
@@ -1,7 +1,13 @@
+import PropTypes from 'prop-types';
 import './style.scss';
 
-const StatsPeriodHeader = ( { children } ) => {
-	return <div className="stats__period-header">{ children }</div>;
+const StatsPeriodHeader = ( { children, className = '' } ) => {
+	return <div className={ `stats__period-header ${ className }`.trim() }>{ children }</div>;
```
Comment (dognose24) [inline client/my-sites/stats/stats-period-header/index.jsx:5]: Could we use `clsx` to combine class names rather than string concat?

**Review by dognose24 — COMMENTED** [review]
Looks good! Some styling would need to be tweaked to align with the design.

|Design|Present|
|-|-|
|截圖 2024-11-14 晚上9 35 24|截圖 2024-11-14 晚上9 56 22|

File: client/my-sites/stats/stats-period-header/index.jsx (hunk 1 above)
Comment (annacmc) [reply client/my-sites/stats/stats-period-header/index.jsx:5]: Yes you're right. The reason I kept this in draft is just as I was working with the classes I discovered that ~7 years ago stats already had a sticky header, and there is a component for this. So I am now investigating that component use to compare :) But if I go with this one, then there does need to be some styling added (the border + padding) and I think we could use `clsx` for combing class names 👍

**Comment by annacmc** [comment]
Closing in favour of https://github.com/Automattic/wp-calypso/pull/96532

**Pull request closed without merging**